src/blueapi/service/handler.py

```diff
@@ -115,6 +115,9 @@ def begin_task(self, task: WorkerTask) -> WorkerTask:
             self._worker.begin_task(task.task_id)
         return task
 
+    def get_tasks_by_status(self, status:str) -> list[TrackableTask[Task]]:
```

Comment:
```suggestion
    def get_tasks_by_status(self, status:TaskStatusEnum) -> list[TrackableTask[Task]]:
```

Reply:
so both handler and worker?

Reply:
Yes, and API

Reply:
` @app.get("/tasks")
def get_tasks(
    status: str = Query("unstarted", description="The status of the tasks to retrieve"),
    handler: BlueskyHandler = Depends(get_handler),
) -> list[TrackableTask]:`

here the string is inevitable, as it is enforced by fastapi. What I'll do is test against the enum or raise an http exception

Reply:
FastAPI allows enums...

Reply:
it doesn't do validation, it's weird to accept enum at the root? https://github.com/tiangolo/fastapi/issues/2910

Reply:
Then there should be a pydantic model:

```python
class TasksRequest(BlueapiBaseModel):
    status: StatusEnum
```

This future-proofs it because future PRs can add additional fields to the model as-and-when required

Reply:
ok, that's actually a great idea, I wish I came up with that